src/controllers/controller.bar.js

```diff
@@ -472,6 +470,47 @@ module.exports = function(Chart) {
 
 			helpers.canvas.unclipArea(chart.ctx);
 		},
+
+		/**
+		 * @private
+		 */
+		_resolveElementOptions: function(rectangle, index) {
+			var me = this;
+			var chart = me.chart;
+			var datasets = chart.data.datasets;
+			var dataset = datasets[me.index];
+			var custom = rectangle.custom || {};
+			var options = chart.options.elements.rectangle;
+			var resolve = helpers.options.resolve;
```

Comment:
I think it'd be cleaner to just use `helpers.options.resolve` directly. It removes a layer of indirection and would also result in smaller minimized code

Reply:
Btw, I love the introduction of the `resolve` helper. The code is so much cleaner now

Reply:
That comes from a copy/paste of the [bubble code](/chartjs/Chart.js/blob/master/src/controllers/controller.bubble.js#L121), where `resolve` is used at multiple places. I would agree if `resolve` was *called* only one time, however I think it's better to keep it cached because it actually removes 2 layers of indirection *at every iteration*.